client/src/components/util/truncate.js

```diff
@@ -0,0 +1,78 @@
+import React, { cloneElement } from "react";
+import { Tooltip, Position } from "@blueprintjs/core";
+
+import { tooltipHoverOpenDelayQuick } from "../../globals";
+
+export default (props) => {
+  const { children } = props;
+  // Truncate only support a single child with a text child
+
+  if (
+    React.Children.count(children) === 1 &&
+    React.Children.count(children.props?.children) === 1
+  ) {
+    const originalString = children.props.children;
+
+    const firstString = originalString.substr(0, originalString.length / 2);
+    const secondString = originalString.substr(originalString.length / 2);
+
+    const splitStyle = {
+      ...children.props.style,
+      display: "flex",
+      overflow: "hidden",
+      justifyContent: "flex-start",
+    };
+
+    const firstStyle = {
+      overflow: "hidden",
+      textOverflow: "ellipsis",
+      whiteSpace: "nowrap",
+      flexShrink: 1,
+      minWidth: 5,
+    };
+
+    const secondStyleInner = {
+      position: "absolute",
+      right: 0,
+      color: "initial",
+    };
+    const secondStyle = {
+      color: "transparent",
+      position: "relative",
+      overflow: "hidden",
+      whiteSpace: "nowrap",
+    };
+
+    const truncatedJSX = (
+      <span style={splitStyle}>
```

Comment:
@seve prefer no variables here, prefer inlining for readability (that's the convention elsewhere in the codebase).